### PR TITLE
feat: allow agent level config with --config flags

### PR DESCRIPTION
**Pull request by GrayFlash**

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -56,6 +57,13 @@ func RunCmd(lg log.Logger, mt *metrics.StatsdMonitor, cfg config.Config) *cobra.
 			"group:core": "true",
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if configFile != "" {
+				var err error
+				cfg, err = config.LoadFromPath(configFile)
+				if err != nil {
+					return err
```
Comment (mabdh) [inline cmd/run.go:64]: why not just

```suggestion
				cfg, err := config.LoadFromPath(configFile)
				if err != nil {
					return err
```
?

File: config/config.go [hunk 2]
```diff
@@ -29,3 +29,15 @@ func Load() (cfg Config, err error) {
```
Comment (mabdh) [inline config/config.go:29]: I think we could also give a config sample `meteor.yaml.sample` in the repo

File: cmd/run.go (hunk 1 above)
Comment (GrayFlash) [reply cmd/run.go:64]: Wouldn’t that make `cfg` a local variable as well??

**Comment by mabdh** [comment]
@GrayFlash perhaps we could also have a new command in Makefile to `copy-config`

```
copy-config:
   cp ./config/meteor.yaml.sample ./meteor.yaml
```

File: config/meteor.yaml.sample [hunk 3]
```diff
@@ -0,0 +1,7 @@
+LOG_LEVEL: hehehehehe
```
Comment (mabdh) [inline config/meteor.yaml.sample:1]: we can make the default to `info` or `debug` 😄 
```suggestion
LOG_LEVEL: debug
```

File: config/meteor.yaml.sample [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+STATSD_ENABLED: false
+STATSD_HOST: "localhost:8005"
+STATSD_PREFIX: lol
```
Comment (mabdh) [inline config/meteor.yaml.sample:4]:
```suggestion
STATSD_PREFIX: meteor
```

File: config/meteor.yaml.sample [hunk 5]
```diff
@@ -0,0 +1,7 @@
+LOG_LEVEL: hehehehehe
+STATSD_ENABLED: false
+STATSD_HOST: "localhost:8005"
```
Comment (mabdh) [inline config/meteor.yaml.sample:3]:
```suggestion
STATSD_HOST: "localhost:8125"
```

8125 is statsd default port

File: config/meteor.yaml.sample (hunk 4 above)
Comment (GrayFlash) [reply config/meteor.yaml.sample:4]: sorry, i wasn’t meant to push it, it was just to test the working 😅😅

File: config/config.go [hunk 6]
```diff
@@ -18,10 +18,17 @@ type Config struct {
 	StopOnSinkError             bool   `mapstructure:"STOP_ON_SINK_ERROR" default:"false"`
 }
 
-func Load() (cfg Config, err error) {
-	err = config.
-		NewLoader(config.WithPath("./")).
-		Load(&cfg)
+func Load(configFile string) (cfg Config, err error) {
+	if configFile != "" {
```
Comment (ravisuhag) [inline config/config.go:22]: @GrayFlash If flag has a default already this check will not be needed.

**Review by mabdh — APPROVED** [review]

**Pull request merged**